### PR TITLE
JIT: check that constant handles are not OK2_CONST_LONGs

### PR DESCRIPTION
**Pull request by AndyAyersMS**
Follow-up from #1788.

Also remove an arm-only handle block that should now be redundant.

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
@briansull PTAL
cc @dotnet/jit-contrib

**Comment by AndyAyersMS** [comment]
The newly added assert is firing. Will dig in and see why.

**Comment by briansull** [comment]
There is a strange code sequence here, that may be relevant:

Assertionprop.cpp line 1341:
```

                /* iconFlags should only contain bits in GTF_ICON_HDL_MASK */
                assert((iconFlags & ~GTF_ICON_HDL_MASK) == 0);
                assertion.op2.u1.iconFlags = iconFlags;
#ifdef _TARGET_64BIT_
                if (op2->TypeGet() == TYP_LONG)
                {
                    assertion.op2.u1.iconFlags |= 1; // Signify that this is really TYP_LONG
                }
#endif // _TARGET_64BIT_
```

**Comment by AndyAyersMS** [comment]
Hmm, I should have checked the assertion unions more carefully -- the iconFlags overlay the top part of the long value, so once an assertion is created, there's no easy way to check if somebody has inadvertently set the icon flags.

I suppose in a debug build we could keep some extra state?

**Comment by AndyAyersMS** [comment]
> I suppose in a debug build we could keep some extra state?

Or perhaps just add padding to `IntVal` on 32 bits, so the icon flags aren't overlaying anything.

**Comment by briansull** [comment]
These flags should not overlap.  I would suggest that we fix that.

**Comment by AndyAyersMS** [comment]
After adding padding, no hits from the new assert in inner or outer loop testing. The padding should not increase the size of the assertion records.

The one remaining failure is in the flaky-seeming trace event test.

File: src/coreclr/src/jit/assertionprop.cpp [hunk 1]
```diff
@@ -1643,6 +1638,13 @@ void Compiler::optDebugCheckAssertion(AssertionDsc* assertion)
         }
         break;
 
+        case O2K_CONST_LONG:
+        {
```
Comment (briansull) [inline src/coreclr/src/jit/assertionprop.cpp]: Don't really need the braces { } here.

**Review by briansull — APPROVED** [review]
Looks Good

File: src/coreclr/src/jit/assertionprop.cpp [hunk 2]
```diff
@@ -1643,6 +1638,13 @@ void Compiler::optDebugCheckAssertion(AssertionDsc* assertion)
...
+            // All handles should be represented by O2K_CONST_INT,
+            // so no handle bits should be set here.
+            assert((assertion->op2.u1.iconFlags & GTF_ICON_HDL_MASK) == 0);
+        }
+
```
Comment (briansull) [inline src/coreclr/src/jit/assertionprop.cpp]: Should have a break here

**Review by briansull — CHANGES_REQUESTED** [review]
I would add a break.

**Review by briansull — APPROVED** [review]
LGTM

**Comment by AndyAyersMS** [comment]
Remaining failures are in the tracevent tests.

**Pull request merged**